src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -255,21 +255,31 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
             }
 
             var resourceTypesString = parsedResourceTypes.Select(x => x.ToString()).ToArray();
-            searchExpressions.AddRange(searchParams.Parameters.Select(
-                    q =>
-                    {
-                        try
-                        {
-                            return _expressionParser.Parse(resourceTypesString, q.Item1, q.Item2);
-                        }
-                        catch (SearchParameterNotSupportedException)
-                        {
-                            unsupportedSearchParameters.Add(q);
 
-                            return null;
-                        }
-                    })
-                .Where(item => item != null));
+            if (resourceTypesString.Length > 1)
+            {
+                InExpression inExpression = new InExpression(FieldName.TokenCode, null, resourceTypesString);
```

Comment:
Does this work now :D?
```suggestion
                Expression inExpression = Expression.In(FieldName.TokenCode, null, resourceTypesString);
```